### DISCUSSION TITLE
HeaderUnits无法找到

### DISCUSSION THREAD

**Discussion by Penguin-SAMA** [xmake-io/xmake#3996] [Q&A]
我直接用的[tests/projects/c++/modules/stl_headerunit_cpp_only](https://github.com/xmake-io/xmake/tree/master/tests/projects/c%2B%2B/modules/stl_headerunit_cpp_only)里的例子，但是无法成功构建，会弹出下面的错误
<img width="416" alt="image" src="https://github.com/xmake-io/xmake/assets/50973632/a64603ec-717b-45a3-86f0-0782594a2e7c">

**Comment by waruqi** [comment]
clang? 如果是 clang 的话，目前它对 stl 的 heaerunits 支持还不完善。

**Comment by Penguin-SAMA** [comment]
gcc13.1
<img width="714" alt="image" src="https://github.com/xmake-io/xmake/assets/50973632/4300ad34-d6fd-4aa4-8dba-c86996c7789d">

**Reply by waruqi** [reply]
贴下完整 -vD logs

xmake -rvD

**Reply by Penguin-SAMA** [reply]
<img width="1279" alt="image" src="https://github.com/xmake-io/xmake/assets/50973632/cbf9f19b-541b-4a27-83c1-eb5b9d400a0b">

**Reply by waruqi** [reply]
mac 上自带的 gcc 其实就是 clang 版本太低。。不支持 c++ header unit 。。。得自己下载 llvm16 用里面的 clang 编译，至少能支持 user header unit 。。但是对 stl header unit 也支持不完善。。

**Reply by Penguin-SAMA** [reply]
我是有另外装官方的gcc13的，我自己手动编译是可以成功编译的，xmake能使用官方的gcc编译器吗

**Reply by waruqi** [reply]
现在里面执行的是 /usr/bin/gcc 。。首先你得确认它是否是 你安装的 gcc13 。。默认它是 apple clang

```
$ /usr/bin/gcc --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/
bin
```

> xmake能使用官方的gcc编译器吗

取决你自己，你让它用，它才会去用

**Reply by Ariaszzzhc** [reply]
> mac 上自带的 gcc 其实就是 clang 版本太低。。不支持 c++ header unit 。。。得自己下载 llvm16 用里面的 clang 编译，至少能支持 user header unit 。。但是对 stl header unit 也支持不完善。。

我 brew 安装的 llvm 17 发现也不能编译 header unit... 使用 xmake f --toolchain=llvm --sdk=/opt/homebrew/opt/llvm -m debug -v 之后进行编译，会提示 header file "pch.h" (aka 'src/pch.h') cannot be imported because it is not known to be a header unit
